src/core/meshcop/dataset_manager.cpp

```diff
@@ -330,14 +330,20 @@ ThreadError DatasetManager::Set(Coap::Header &aHeader, Message &aMessage, const
     Tlv::Type type;
     bool isUpdateFromCommissioner = false;
     bool isUpdateAffectConnectivity = false;
+    bool isUpdateIdenticalValue = true;
```

Comment:
We could simplify the implementation by combining the two bool variables (isUpdateAffectConnectivity and isUpdateIdenticalValue), use a single bool variable to indicate that there are affect connectivity tlvs and the values are different from its own.